Sources/Rendering/Core/CellPicker/index.js

```diff
@@ -317,17 +335,20 @@ function vtkCellPicker(publicAPI, model) {
     const x1 = [0, 0, 0];
     const x2 = [0, 0, 0];
     for (let i = 0; i < 3; i++) {
-      x1[i] = (p1[i] - origin[i]) / spacing[i];
-      x2[i] = (p2[i] - origin[i]) / spacing[i];
+      x1[i] = (q1[i] - origin[i]) / spacing[i];
+      x2[i] = (q2[i] - origin[i]) / spacing[i];
     }
     const x = [0, 0, 0, 0];
     const xi = [0, 0, 0];
 
     const sliceSize = dims[1] * dims[0];
     const rowSize = dims[0];
-    const nSteps = 100;
+    // here the step is the 1 over the distance between volume index location x1 and x2
+    const step = 1 / Math.sqrt(vtkMath.distance2BetweenPoints(x1, x2));
```

Comment:
shouldn't step be not smaller than something that correspond to the size of a voxel ?

Reply:
This step is related to the percentage of the size of the distance between x1 and x2. It increments t the parametric value